Formula/newsboat.rb

```diff
@@ -14,29 +15,96 @@ class Newsboat < Formula
     sha256 x86_64_linux:  "a8e76829cb365ae43de0e3aa6d442c1a95378abb59a19f6a2f006f0910e3d44b"
   end
 
-  deprecate! date: "2021-11-26", because: "libstfl is deprecated"
-
   depends_on "asciidoctor" => :build
   depends_on "pkg-config" => :build
+  depends_on "python@3.10" => [:build, :test]
   depends_on "rust" => :build
+  depends_on "swig" => :build
   depends_on "xz" => :build
   depends_on "gettext"
   depends_on "json-c"
-  depends_on "libstfl"
 
+  depends_on "ruby"
+
+  uses_from_macos "perl" => [:build, :test]
   uses_from_macos "curl"
   uses_from_macos "libxml2"
   uses_from_macos "libxslt"
+  uses_from_macos "ncurses"
   uses_from_macos "sqlite"
 
+  # Newsboat have their own libstfl fork. Upstream libsftl is gone:
+  # https://github.com/Homebrew/homebrew-core/pull/89981
+  # They do not want to be the new upstream, but use that fork as a temporary
+  # workaround until they migrate to some rust crate
+  # https://github.com/newsboat/newsboat/issues/1881
+  resource("libstfl") do
+    url "https://github.com/newsboat/stfl.git",
+        revision: "c2c10b8a50fef613c0aacdc5d06a0fa610bf79e9"
+  end
+
   def install
+    resource("libstfl").stage do
+      if OS.mac?
+        ENV.append "LDLIBS", "-liconv"
+        ENV.append "LIBS", "-lncurses -lruby -liconv"
+
+        inreplace "stfl_internals.h", "ncursesw/ncurses.h", "ncurses.h"
+        inreplace %w[stfl.pc.in ruby/Makefile.snippet], "ncursesw", "ncurses"
+
+        inreplace "Makefile" do |s|
+          s.gsub! "ncursesw", "ncurses"
+          s.gsub! "-Wl,-soname,$(SONAME)", "-Wl"
+          s.gsub! "libstfl.so.$(VERSION)", "libstfl.$(VERSION).dylib"
+          s.gsub! "libstfl.so", "libstfl.dylib"
+        end
+
+        # Fix ncurses linkage for Perl bundle
+        inreplace "perl5/Makefile.PL", "-lncursesw", "-L#{MacOS.sdk_path}/usr/lib -lncurses"
+      else
+        ENV.append "LIBS", "-lncursesw -lruby"
+        inreplace "Makefile", "$(LDLIBS) $^", "$^ $(LDLIBS)"
+      end
+
+      ENV.prepend_path "PATH", Formula["python@3.10"].opt_libexec/"bin"
+      xy = Language::Python.major_minor_version "python3"
+      py_cflags = Utils.safe_popen_read("python-config", "--cflags").chomp
+
+      inreplace "python/Makefile.snippet" do |s|
+        # Install into the site-packages in the Cellar (so uninstall works)
+        s.change_make_var! "PYTHON_SITEARCH", lib/"python#{xy}/site-packages"
```

Comment:
Should this be in `libexec`? Does `newsboat` even need the Python bindings?

Reply:
You are right, I removed the whole bindings part. It just needs the .so file.